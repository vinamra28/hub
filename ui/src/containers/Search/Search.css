.hub-search {
  border-radius: 4px;
  background: white;
  border: 0px;
  font-style: italic;
  color: black;
  width: 20em;
  outline: none;
}
