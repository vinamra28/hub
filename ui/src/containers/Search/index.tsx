import { TextInput } from '@patternfly/react-core';
import React, { useState } from 'react';
import './Search.css';

const Search: React.FC = () => {
  const [value, setValue] = useState('');

  const setInput = (event: string) => {
    setValue(event);
  };

  return (
    <TextInput
      value={value}
      type="search"
      onChange={setInput}
      aria-label="text input example"
      placeholder="Search for resources..."
      spellCheck="false"
      className="hub-search"
    />
  );
};

export default Search;
