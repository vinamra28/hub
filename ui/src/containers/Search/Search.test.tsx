import React from 'react';
import renderer from 'react-test-renderer';
import Search from '.';

it('should render the search component', () => {
  const tree = renderer.create(<Search />).toJSON();
  expect(tree).toMatchSnapshot();
});
