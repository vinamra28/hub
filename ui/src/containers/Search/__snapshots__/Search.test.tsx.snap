// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the search component 1`] = `
<input
  aria-invalid={false}
  aria-label="text input example"
  className="pf-c-form-control hub-search"
  disabled={false}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  placeholder="Search for resources..."
  readOnly={false}
  required={false}
  spellCheck="false"
  type="search"
  value=""
/>
`;
